contributors/design-proposals/volume-provisioning.md

```diff
@@ -412,9 +419,13 @@ type StorageClass struct {
   // Parameters for dynamic volume provisioner.
   Parameters map[string]string `json:"parameters,omitempty"`
 }
-
 ```
 
+StorageClass can have annotation
+`volume.beta.kubernetes.io/storage-class-selectors` that contains free-form
+english text with description of labels that are supported by the provisioner
+in pvc.spec.selector when provisioning a new PV for this StorageClass.
```

Comment:
this should be a structured, first-class field on the storage class, so that clients can be built to allow users to specify these

straw man type:
```
SelectorOptions []SelectorComponent

type SelectorComponent struct {
  // Label selector key
  Key string

  // If unspecified in the PVC, this value will be used
  DefaultValue string
  // List of all allowed values. Either AllowedValues or DisallowedValues may be specified, not both. If neither are specified, any valid label value is allowed.
  AllowedValues []SelectorValue
  // List of explicitly disallowed values. Either AllowedValues or DisallowedValues may be specified, not both. If neither are specified, any valid label selector value is allowed.
  DisallowedValues []SelectorValue

  // Optional display name. May not contain newlines. If unspecified, Key is used.
  DisplayName string
  // Optional description. May contain newlines and links.
  Description string
}

type SelectorValue struct {
  // Selector value
  Value string
  // Optional display name. May not contain newlines. If unspecified, Key is used.
  DisplayName string
  // Optional description. May contain newlines and links.
  Description string
}
```

that could look something like: 
```
...
"selectorOptions": [
  {
    "key":"zone",
    "defaultValue":"east",
    "allowedValues": [{"value":"east"},{"value":"west"}],
    "displayName":"Availability zone"
  },
  {
    "key":"type",
    "defaultValue":"hdd",
    "allowedValues": [
      {"value":"ssd", "displayName":"Solid-state"},
      {"value":"hdd", "displayName":"Spinning-disk"}
    ],
    "displayName":"Storage type"
  }
]
...
```